docs/src/appendix/packaging.md

```diff
@@ -0,0 +1,501 @@
+# Packaging
+
+This appendix describes the package format consumed and produced by the Miden toolchain.
+
+## Overview
+
+A Miden package represents a compiled _program_ or _library_, with sufficient metadata to
+allow any Miden VM instance to execute the code it contains. Packages may reference other
+packages, and all packages in the dependency graph of a program must be available to the
+VM in order for the code to be executed. More about package dependencies will be covered
+later in the [Dependencies](#dependencies) section.
+
+A package consists of the following:
+
+* A manifest that provides top-level metadata about the package
+* The Merkelized Abstract Syntax Tree (MAST) which was produced by compiling the original
+source code. The MAST is what will be directly executed by the Miden VM at runtime.
+* A WebAssembly Interface Types (WIT) component definition, which is used to describe
+the interfaces exported from the package, their type signatures, and other useful metadata
+about the structure of the package. See the [Type Descriptor](#type-descriptor)
+section for more details.
+* Zero or more read-only data segments. See the [Read-Only Data Segments](#read-only-data-segments)
+* One or more optional items that can be used with the package for development
+and debugging, the structure of which depends on the specific item:
+  * Debug info
+  * Documentation
```

Comment:
So that's true today, but in the future, particularly if we develop some runtime aspects of the Component Model, the definition of a component is essential for that.

More broadly though, the component definition (what you are referring to here as WIT, which is a bit of a misnomer, WIT is primarily the textual format for defining WebAssembly components) tells us what components a package provides, and how to instantiate them. For now, we don't have any runtime support for components, so this largely just tells us what the APIs are of the package in terms that enable interop with that package - but this will change as we support more and richer functionality.

In other words, I expect that eventually the VM will use a component definition to inform/manage certain aspects of instantiation/initialization of the components described by a package, even if that is not the case today.